pkg/console/controllers/route/controller.go

```diff
@@ -130,17 +134,37 @@ func (c *RouteSyncController) sync() error {
 	}
 
 	updatedOperatorConfig := operatorConfig.DeepCopy()
+	updateConditionFuncs := []v1helpers.UpdateStatusFunc{}
 
 	defaultRouteErrReason, defaultRouteErr := c.SyncDefaultRoute(updatedOperatorConfig)
-	status.HandleProgressingOrDegraded(updatedOperatorConfig, "DefaultRouteSync", defaultRouteErrReason, defaultRouteErr)
+	defaultRouteSyncConditions := status.HandleProgressingOrDegraded("DefaultRouteSync", defaultRouteErrReason, defaultRouteErr)
+	updateConditionFuncs = append(updateConditionFuncs, defaultRouteSyncConditions...)
```

Comment:
Definitely want to eliminate the need for the added conditional logic everywhere.  I think we can tuck most of this away in our own status helpers, and perhaps just wrap the `v1helpers.UpdateStatus` call with our own (perhaps existing) helpers and not change our controllers much.